### PR TITLE
Feature/multi stage dockerfile

### PR DESCRIPTION
**Pull request by porsche-rbieniek**
Please ensure that your pull request adheres to our [contribution guidelines](https://github.com/oss-review-toolkit/ort/blob/master/CONTRIBUTING.md). Thank you!

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -133,55 +258,73 @@ RUN /opt/ort/bin/import_proxy_certs.sh && \
       /opt/ort/bin/import_certificates.sh /tmp/certificates/; \
     fi && \
     # Install VCS tools (no specific versions required here).
+    mkdir -p /usr/local/bin && \
+    if [ ! -f /usr/bin/python ]; then \
+       ln -s /usr/bin/python3 /usr/bin/python; \
```
Comment (maxhbr) [inline Dockerfile:263]: I think the better alternative is `apt-get install python-is-python3`

File: Dockerfile [hunk 2]
```diff
@@ -49,43 +79,138 @@ RUN --mount=type=cache,target=/tmp/.gradle/ \
...
+        tar -Jx -f - -C /opt/scancode/ && \
+        # Trigger ScanCode configuration for Python 3 and reindex licenses initially.
+        ( cd /opt/scancode/scancode-toolkit-${SCANCODE_VERSION}; PYTHON_EXE=/usr/bin/python3.6 ./scancode --reindex-licenses ) && \
+        chmod -R o=u /opt/scancode/scancode-toolkit-${SCANCODE_VERSION} && \
+        ln -s /opt/scancode/scancode-toolkit-${SCANCODE_VERSION}/scancode /usr/local/bin/scancode
+
+FROM base-image as android-sdk-buiild
+
+ARG CRT_FILES
+ARG ANDROID_SDK_VERSION
```
Comment (maxhbr) [inline Dockerfile:160]: There is no default? Or is that pulled in from a previous stage?

File: Dockerfile [hunk 3]
```diff
@@ -49,43 +79,138 @@ RUN --mount=type=cache,target=/tmp/.gradle/ \
...
     HASKELL_STACK_VERSION=2.1.3 \
     NPM_VERSION=7.20.6 \
-    PYTHON_PIPENV_VERSION=2018.11.26 \
-    PYTHON_VIRTUALENV_VERSION=15.1.0 \
+    PYTHON_PIPENV_VERSION=2021.5.29 \
+    PYTHON_VIRTUALENV_VERSION=20.2.0 \
+    PYTHON_FUTURE_VERSION=0.18.2 \
     SBT_VERSION=1.3.8 \
-    YARN_VERSION=1.22.10 \
+    YARN_VERSION=1.22.4 \
```
Comment (maxhbr) [inline Dockerfile:195]: downgrade?

File: Dockerfile [hunk 4]
```diff
@@ -133,55 +258,73 @@ RUN /opt/ort/bin/import_proxy_certs.sh && \
...
-    rm /opt/ort.tar && \
-    /opt/ort/bin/ort requirements
+    rm /opt/ort.tar
+    # Disabled to due hardcoded scancode version in requirements check
+    # /opt/ort/bin/ort requirements
+
+COPY --from=java-build /usr/local/src/ort/helper-cli/build/scripts/orth /opt/ort/bin/
+COPY --from=java-build /usr/local/src/ort/helper-cli/build/libs/helper-cli-*.jar /opt/ort/lib/
+
+COPY curations.tar /
```
Comment (maxhbr) [inline Dockerfile:325]: `ADD` can combine a `COPY` and extract: https://docs.docker.com/engine/reference/builder/#add

Comment (nicorikken) [reply Dockerfile:325]: I tried building this container, but it fails on this COPY as the curations.tar isn't available (anymore).

**Comment by sschuberth** [comment]
By now, it was decided to go with https://github.com/oss-review-toolkit/ort/pull/5830 instead.

**Pull request closed without merging**